src/python/pants/backend/helm/dependency_inference/subsystem.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from enum import Enum
+
+from pants.option.option_types import EnumOption
+from pants.option.subsystem import Subsystem
+from pants.util.strutil import softwrap
+
+
+class UnownedDependencyUsage(Enum):
+    """What action to take when an inferred dependency is unowned."""
+
+    RaiseError = "error"
+    LogWarning = "warning"
+    DoNothing = "ignore"
+
+
+class UnownedDependencyError(Exception):
+    """The inferred dependency does not have any owner."""
+
+
+class HelmInferSubsystem(Subsystem):
+    options_scope = "helm-infer"
+    help = "Options controlling which dependencies will be inferred for Helm targets."
+
+    unowned_dependency_behavior = EnumOption(
+        default=UnownedDependencyUsage.LogWarning,
```

Comment:
Something in me would really like for this default to be `error` but the error (and warning) message should, imo, contain a reference to the official docs where how to safely silence the error, i.e. changing this option to be `warning`/`ignore`, or  listing the external docker images if we go down that path.

Reply:
I think that makes sense, so we'd never continue even though we're not sure everything is correct. I think if we implemented your suggestion for `third_party_docker_images` we'd be able to say that the "default assumption" is that passing in images like this is for local images, and a use should make it clear that it isn't local either by using a full name or by adding it to the list of known 3rd party images

Reply:
yeah, and probably `third_party_docker_images` is not the right name but the point of having the inference error or warn would be very helpful for a case in which targets are relocated inside the repo (and therefore their address change) bur then forgot to update the references in the `helm_deployment` targets.

It also kind of contributes to the idea of having hermetic builds. Although for that to be actually hermetic we would need to homage image digests or something similar (as it is done with the external downloaded tools)

Reply:
how about `external_docker_images`?
Also changing the default behaviour to "error" broke a few tests :P The python backend has the default of "warning". Do we still want to have the default be "error"? (I added a note in the error on the setting to change the behaviour, so it should be easy for someone to change)